onert-micro/luci-interpreter/src/kernels/Pool2DCommon.h

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright (c) 2024 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef LUCI_INTERPRETER_KERNELS_POOL2DCOMMON_H
+#define LUCI_INTERPRETER_KERNELS_POOL2DCOMMON_H
+
+#include "Builders.h"
+#include "SISOKernel.h"
+#include "Utils.h"
+
+namespace luci_interpreter
+{
+
+inline void configure_kernel_CirclePool2DCommon(const circle::Operator *cur_op,
+                                                BaseRuntimeGraph *runtime_graph)
+{
+  const kernels::SISOKernel siso_kernel(cur_op, runtime_graph);
+
+  LUCI_INTERPRETER_CHECK(Tensor::element_type(siso_kernel.input()) ==
+                         Tensor::element_type(siso_kernel.output()));
+  LUCI_INTERPRETER_CHECK(Tensor::num_dims(siso_kernel.input()) == 4);
+  LUCI_INTERPRETER_CHECK(Tensor::num_dims(siso_kernel.input()) ==
+                         Tensor::num_dims(siso_kernel.output()));
+
+#ifndef DIS_QUANT
+  if (Tensor::element_type(siso_kernel.input()) == DataType::U8)
+  {
+    LUCI_INTERPRETER_CHECK(
+      std::abs(Tensor::scale(siso_kernel.output()) - Tensor::scale(siso_kernel.input())) <= 1.0e-6);
+    LUCI_INTERPRETER_CHECK(Tensor::zero_point(siso_kernel.output()) ==
+                           Tensor::zero_point(siso_kernel.input()));
+  }
+  else if (Tensor::element_type(siso_kernel.input()) == DataType::S16)
+  {
+    LUCI_INTERPRETER_CHECK(
+      std::abs(Tensor::scale(siso_kernel.output()) - Tensor::scale(siso_kernel.input())) <= 1.0e-6);
+    LUCI_INTERPRETER_CHECK(Tensor::zero_point(siso_kernel.input()) == 0 &&
+                           Tensor::zero_point(siso_kernel.output()) == 0);
+  }
+#endif // DIS_QUANT
+}
+
+inline luci_interpreter_pal::PoolParams createPoolParams(const circle::Operator *cur_op,
+                                                         BaseRuntimeGraph *runtime_graph)
```

Comment:
I think it's better to pass `const kernels::SISOKernel &siso_kernel` parameter here so as not to read the input and output tensor twice, since this has already been done in the calling function.
What do you think?)

Reply:
Do you mean to pass `const kernels::SISOKernel &siso_kernel` instead of `BaseRuntimeGraph *runtime_graph`? It looks better.